.github/workflows/nightly.yml

```diff
@@ -0,0 +1,72 @@
+on:
+  # runs on 4:30 AM PST or 5:30 AM PDT
+  schedule:
+    - cron:  '30 12 * * *'
+jobs:
+  build-and-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Setup Java 9
+        uses: actions/setup-java@v1.4.3
+        with:
+          java-version: '9'
+          java-package: jdk
+          architecture: x64
+      - name: set JDK_9 environment variable for kotlin compiler
+        shell: bash
+        env:
+          ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+        run: echo ::set-env name=JDK_9::$(echo $JAVA_HOME)
+      - name: Setup Java 11
+        uses: actions/setup-java@v1.4.3
+        with:
+          java-version: '11'
+          java-package: jdk
+          architecture: x64
+
+      # Checkout
+      - uses: actions/checkout@v2
```

Comment:
This tests all three OS on the main branch. There are 9 combinations though:
{main, next-release, current-release} x {windows, mac, linux}

where {main, current-release} x {linux, mac} are tested on pushes (main.yml) already. Can you make this a matrix and skip {main, current-release} x {linux, mac}?